### ISSUE TITLE
Increase Selectionw works only in C# files

### ISSUE THREAD

**Issue by martinsura** [justcla/HotCommands#58]
Hi,

Fantastic extensions with one small problem. Ctrl+w works only in C# files.

Is here any workaround how to use ctrl+w for C# class and classic ctrl+w in other files(html,css)

Thanks

**Comment by photex** [comment]
Attempting to use 'Edit.IncreaseSelection' in a c++ file pops up an error. Would love to be able to use this.

**Comment by tamlin-mike** [comment]
@photex Context-free C++ (and in fact any language with similar syntax) widening is implemented in https://github.com/tamlin-mike/HotCommands/blob/tamlin-dev/HotCommands/Commands/ProjectLess/NoProj_ExpandSelection.cs

(I should probably tidy this up and submit a PR)

**Comment by johndebord** [comment]
Any update on this issue?
I as well get a pop-up indicating "One or more errors occurred" when trying to expand/shrink selection in C++ files. This is one of my favorite commands in other editors.

**Comment by justcla** [comment]
Sorry - I seem to have been ignoring this thread. My apologies.
I don't think I'll ever program HotCommands to expand C++ files. I simply don't know how to do it. C++ does not use Roslyn. I could reach out to the C++ guys and see if they want to give it a go. But I'm hoping they will simply build it into the product soon.
I can tell you that we've created a basic C# implementation in VS2017 v15.5. So we'll probably be retiring the ExpandSelection feature from HotCommands (at least for VS2017 versions).

I note that Amadeusz has pushed a PR that prevents the error when you try to expand selection in C++.
I doesn't provide the functionality you want, but at least it will stop the errors. :-)

In the meantime, I'll push the C++ guys to see what they can do for us.

**Comment by johndebord** [comment]
How about you hire me and I'll do it! :)
Would be awesome to work at Microsoft.

**Comment by justcla** [comment]
Well, if you can make some good PRs to HotCommands, I'm sure it will look good on an application. ;-)

**Comment by johndebord** [comment]
Challenge accepted!

**Comment by tamlin-mike** [comment]
> In the meantime, I'll push the C++ guys to see what they can do for us.

See #53.

TL; DR; Nada. Microsoft is as uninterested in providing infrastructure in this area as they were back in the 90's.

**Comment by justcla** [comment]
I haven't given up hope. I'm sure they'll get to it soon. :-)

**Comment by nilzen** [comment]
@tamlin-mike can your code expand on attributes in html or is it just for C++? Anyway a PR would be awesome!

**Comment by justcla** [comment]
My extension does expand in HTML. I think it only works in C# (as it uses the Roslyn Syntax Tree).

The in-product version will work in HTML, but is very clunky.
We need the Web Tools PM to make changes for it to work in HTML.
Fortunately...I am not the Web Tools PM. :-)
So I will see if I can get Expand Selection working perfectly in HTML by the time we officially release the next version of Visual Studio.
(I need it working as much as you do, so I'm pretty keen to push it.) ;-)
